From e8935dc20bb0822570fb33e4756dbb9cc30b6b1e Mon Sep 17 00:00:00 2001
From: jakub <jakub@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Wed, 1 Oct 2014 21:01:39 +0000
Subject: [PATCH 278/326] 	PR debug/63342 	* dwarf2out.c
 (loc_list_from_tree): Handle TARGET_MEM_REF and 	SSA_NAME.

	* gcc.dg/pr63342.c: New test.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_8-branch@215784 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog                  |  4 ++++
 gcc/dwarf2out.c                |  4 ++++
 gcc/testsuite/ChangeLog        |  3 +++
 gcc/testsuite/gcc.dg/pr63342.c | 26 ++++++++++++++++++++++++++
 4 files changed, 37 insertions(+)
 create mode 100644 gcc/testsuite/gcc.dg/pr63342.c

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index 23bfb74..8b5d629 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,5 +1,9 @@
 2014-10-01  Jakub Jelinek  <jakub@redhat.com>
 
+	PR debug/63342
+	* dwarf2out.c (loc_list_from_tree): Handle TARGET_MEM_REF and
+	SSA_NAME.
+
 	PR target/63428
 	* config/i386/i386.c (expand_vec_perm_pshufb): Fix up rperm[0]
 	argument to avx2_permv2ti.
diff --git a/gcc/dwarf2out.c b/gcc/dwarf2out.c
index f387e61..c68453e 100644
--- a/gcc/dwarf2out.c
+++ b/gcc/dwarf2out.c
@@ -13882,6 +13882,10 @@ loc_list_from_tree (tree loc, int want_address)
       have_address = 1;
       break;
 
+    case TARGET_MEM_REF:
+    case SSA_NAME:
+      return NULL;
+
     case COMPOUND_EXPR:
       return loc_list_from_tree (TREE_OPERAND (loc, 1), want_address);
 
diff --git a/gcc/testsuite/ChangeLog b/gcc/testsuite/ChangeLog
index 6398f2f..ac676af 100644
--- a/gcc/testsuite/ChangeLog
+++ b/gcc/testsuite/ChangeLog
@@ -1,5 +1,8 @@
 2014-10-01  Jakub Jelinek  <jakub@redhat.com>
 
+	PR debug/63342
+	* gcc.dg/pr63342.c: New test.
+
 	PR target/63428
 	* gcc.dg/torture/vshuf-4.inc: Move test 122 from EXPTESTS
 	to test 24 in TESTS.
diff --git a/gcc/testsuite/gcc.dg/pr63342.c b/gcc/testsuite/gcc.dg/pr63342.c
new file mode 100644
index 0000000..499d508
--- /dev/null
+++ b/gcc/testsuite/gcc.dg/pr63342.c
@@ -0,0 +1,26 @@
+/* PR debug/63342 */
+/* { dg-do compile } */
+/* { dg-options "-g -O2" } */
+/* { dg-additional-options "-fpic" { target fpic } } */
+
+static __thread double u[9], v[9];
+
+void
+foo (double **p, double **q)
+{
+  *p = u;
+  *q = v;
+}
+
+double
+bar (double x)
+{
+  int i;
+  double s = 0.0;
+  for (i = 0; i < 9; i++)
+    {
+      double a = x + v[i];
+      s += u[i] * a * a;
+    }
+  return s;
+}
-- 
2.0.3

